### PR TITLE
locale update and script improvements

### PR DESCRIPTION
**Pull request by lzap**

### CODE REVIEW

**Comment by lzap** [comment]
Okay one thing - if you now fully follow 

http://projects.theforeman.org/projects/foreman/wiki/Translating#How-to-pull-translations

it does use GNU gettext's msgfmt to merge PO files, which leads to wrapping. I like wrapped version more. If you are not against, I'd rather keep that.

I think it might be better to squash updting and merging commits together because one is after tx pull is done and one is after merge.

On the other hand, I might want to skip msgfmt completely, as from tx it will be always coming in this format.

**Comment by lzap** [comment]
Ok squashed and removed that extra po-update.

**Comment by domcleal** [comment]
@lzap I don't mind how we do it, but I have some concerns:

a) JavaScript locale files haven't been updated, you must run the rake task after updating the .po files.  I used to pull the po files and run "rake gettext:find" to both pull translations and extract new strings.  It looks like you've only pulled new translations and haven't updated the JS locale files either.

b) as above, no new extractions?

c) some of the fuzzy matches have disappeared, is that deliberate?  As an example at the top of the de .po file "%{value} gehört nicht zum Betriebssystem %{os}".

**Comment by lzap** [comment]
a) Yeah I apparently forgot to:

```
be rake gettext:po_to_json && git commit -a -m "i18n - updating javascript translations"
```

b) yeah I skipped that, fixing

c) When we pull a po file from Transifex fuzzy translations are never included because they are not valid translations. Transifex does keep the fuzzy msgstr's as "suggestions" to the translator. We are not doing msgmerge after pulling translations - it is not needed as transifex does this for us.

Ok I reviewed the workflow and what we have on the wiki and changed that once again. Check out http://projects.theforeman.org/projects/foreman/wiki/Translating#How-to-pull-translations now.

This gives us nicer patches, note the second commit is still ugly, because there was a msg merge last time. Next update will be much cleaner according these instructions.

**Comment by domcleal** [comment]
:+1: generally, this seems to work well.  Could we just do all three steps in a single commit, maybe run it completely from the makefile?  There's not much value in three commits each time we do this.

**Comment by lzap** [comment]
Hmmm I can squash in two, but as same files are modified twice (and then reverted), I'd rather keep this separate.

**Comment by lzap** [comment]
All reworked. Now one command rules them all.

File: locale/Makefile [hunk 1]
```diff
@@ -44,13 +44,19 @@ uniq-po:
...
-	-git commit -a -m 'i18n - merging gettext po files'
+update-rails:
+	../script/update-rails-locales.sh
+
+extract-strings:
+	bundle exec rake locale:find
+
+tx-update: update-rails tx-pull extract-strings
+	# merging po files is unnecessary when using transifex.com
+	git checkout -- ../locale/*/*po
```
Comment (domcleal) [inline locale/Makefile]: The newly pulled translations don't get committed now.

Comment (lzap) [inline locale/Makefile]: They do, I have hidden one additional Makefile change in the last
commit. I split that, now you should see that.

I tested that, it works fine.

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

Comment (domcleal) [inline locale/Makefile]: I'm not sure I see... but I also don't see any new .po files updated in this PR, shouldn't I?

File: locale/Makefile [hunk 2]
```diff
@@ -44,13 +44,19 @@ uniq-po:
...
+	git commit -a -m "i18n - extracting new, updating rails, pulling from tx"
```
Comment (lzap) [inline locale/Makefile]: Over here ^^^

I think you misunderstood me, PO merging is not necessary anymore. We are not doing that. That is the reason why we throw the changes out (it is not possible to tell to ruby gettext not to do this step).

**Comment by lzap** [comment]
:facepalm: Fixed.

**Comment by domcleal** [comment]
Thanks @lzap :)  Merged the first two commits as 836679f, 9dc1001and will run the new make task once we've finished the 1.3 features.

**Pull request closed without merging**